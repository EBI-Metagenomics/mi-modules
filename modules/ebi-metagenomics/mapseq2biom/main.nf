
process MAPSEQ2BIOM {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"

    input:
    tuple val(meta), path(msq)
    tuple path(db_fasta), path(db_tax), path(db_otu), path(db_mscluster), val(db_label)

    output:
    tuple val(meta), path("${meta.id}.txt")                                 , emit: krona_input
    tuple val(meta), path("${meta.id}.tsv"), path("${meta.id}.notaxid.tsv") , emit: biom_out
    path "versions.yml"                                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"


    """
    mapseq2biom \
        $args
        --krona ${prefix}.txt \
        --no-tax-id-file ${prefix}.notaxid.tsv \
        --label $label \
        --query ${mapseq_out} \
        --otu-table ${db_otu} \
        --out-file ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mapseq2biom: 0.1.0
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}.txt
    touch ${prefix}.notaxid.tsv
    touch ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mapseq2biom: 0.1.0
    END_VERSIONS
    """
}
