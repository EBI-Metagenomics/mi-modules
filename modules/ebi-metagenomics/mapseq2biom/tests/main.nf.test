
nextflow_process {

    name "Test Process MAPSEQ2BIOM"
    script "../main.nf"
    process "MAPSEQ2BIOM"
    config "./nextflow.config"
    
    tag "modules"
    tag "modules_nfcore"
    tag "mapseq2biom"

    test("test_silva - mseq") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test.mseq", checkIfExists: true)
                    ]
                input[1] = [
                    file("${moduleDir}/tests/silva_ssu/test.fasta", checkIfExists: true),
                    file("${moduleDir}/tests/silva_ssu/test.txt", checkIfExists: true),
                    file("${moduleDir}/tests/silva_ssu/test.otu", checkIfExists: true),
                    file("${moduleDir}/tests/silva_ssu/test.mscluster", checkIfExists: true),
                    "SILVA-SSU"
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert snapshot(process.out.krona_input).match("krona_input") },
                { assert snapshot(process.out.biom_out).match("biom_out") },
                { assert snapshot(process.out.biom_out).match("biom_notaxid_out") }
            )
        }

    }

    test("test_silva - mseq - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/test.mseq", checkIfExists: true)
                    ]
                input[1] = [
                    file("${moduleDir}/tests/silva_ssu/test.fasta", checkIfExists: true),
                    file("${moduleDir}/tests/silva_ssu/test.txt", checkIfExists: true),
                    file("${moduleDir}/tests/silva_ssu/test.otu", checkIfExists: true),
                    file("${moduleDir}/tests/silva_ssu/test.mscluster", checkIfExists: true),
                    "SILVA-SSU"
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert snapshot(process.out.krona_input).match("krona_input_stub") },
                { assert snapshot(process.out.biom_out).match("biom_out_stub") },
                { assert snapshot(process.out.biom_out).match("biom_notaxid_out_stub") }
            )
        }

    }

}
