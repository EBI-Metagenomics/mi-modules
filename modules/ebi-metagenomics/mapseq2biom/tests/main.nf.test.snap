{
    "test_silva - mseq - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,fa3ef8980608011a1ecdd15f93889c0a"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.notaxid.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d8270f637a909a58d0070c3abc946c52"
                ],
                "biom_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.notaxid.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "krona_input": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,fa3ef8980608011a1ecdd15f93889c0a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d8270f637a909a58d0070c3abc946c52"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-10T15:52:50.670433"
    },
    "krona_input": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.txt:md5,fa3ef8980608011a1ecdd15f93889c0a"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-10T16:01:39.324767"
    },
    "biom_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.tsv:md5,627ca596232e79fe2777f0e9352df993",
                    "test.notaxid.tsv:md5,4b4dc1c98f5da99339fa477dd47f371d"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-10T16:01:39.333509"
    },
    "test_silva - mseq": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,fa3ef8980608011a1ecdd15f93889c0a"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,627ca596232e79fe2777f0e9352df993",
                        "test.notaxid.tsv:md5,4b4dc1c98f5da99339fa477dd47f371d"
                    ]
                ],
                "2": [
                    "versions.yml:md5,d8270f637a909a58d0070c3abc946c52"
                ],
                "biom_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,627ca596232e79fe2777f0e9352df993",
                        "test.notaxid.tsv:md5,4b4dc1c98f5da99339fa477dd47f371d"
                    ]
                ],
                "krona_input": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,fa3ef8980608011a1ecdd15f93889c0a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d8270f637a909a58d0070c3abc946c52"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-10T16:01:39.312655"
    },
    "biom_out_stub": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "test.notaxid.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-10T16:09:22.899999"
    },
    "krona_input_stub": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.txt:md5,fa3ef8980608011a1ecdd15f93889c0a"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-10T16:09:22.893604"
    }
}